### ISSUE TITLE
Crash when adding icon to sync now setting

### ISSUE THREAD

**Issue by tarikeshaq** [mozilla-mobile/firefox-ios#16644]
## Steps to reproduce
* This is fairly hard to reproduce unless you have a broken account to start with, I came across this debugging an fxa issue
* [the `addIcon` function in the Sync Now setting](https://github.com/mozilla-mobile/firefox-ios/blob/0c9237af555a862461f3722f5978bda30ab82e6a/Client/Frontend/Settings/Main/Account/SyncNowSetting.swift#L240-L253) can crash with the following error:

```
2023-09-28 13:02:44.980118-0400 Client[84398:32774897] *** Terminating app due to uncaught exception 'NSGenericException', reason: 'Unable to activate constraint with anchors <NSLayoutXAxisAnchor:0x60000217af00 "UITableViewLabel:0x7f7fad871680.leading"> and <NSLayoutXAxisAnchor:0x60000217ac00 "UIImageView:0x7f7facf8d980.trailing"> because they have no common ancestor.  Does the constraint or its anchors reference items in different view hierarchies?  That's illegal.'
```

### Expected behavior
* A warning symbol is displayed next to the `Sync Now` text label

### Actual behavior
* The application crashes

### Device & build information
* Device: ?
* Build version: ?

### Notes
* I'm happy to provided my test account that is broken and causes this behavior
Attachments:

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-7498)

**Comment by dnarcese** [comment]
@tarikeshaq what causes the account to be broken?

**Comment by tarikeshaq** [comment]
A specific sequence of events in https://bugzilla.mozilla.org/show_bug.cgi?id=1855264, it's unlikely in practice and requires rapid switching between accounts in desktop.

But the crash here is on the settings menu in the code path where we add a warning icon - once a user is in this state, everytime they open their settings menu the application crashes

**Comment by data-sync-user** [comment]
➤ Stefan Vladut commented:

Tarik Eshaq I wasn’t able to reproduce this, could I get the already broken account?

**Comment by data-sync-user** [comment]
➤ Tarik Eshaq commented:

Hi Stefan Vladut ! I Apologise for the delay, just got back from PTO - I can send you the account information to reproduce, where’s the best place to send it (are you available on Slack, etc?) It’s a test account so no harm in sharing credentials just need to do so in a private forum (as this syncs to GitHub)

**Comment by data-sync-user** [comment]
➤ Stefan Vladut commented:

hello Tarik Eshaq , welcome back! No worries about the delay as I had stuff to do in other parts of the app, I can be found on slack as svladut or email at svladut@mozilla.com

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on v119.2 (35843) with iPhone 13 Pro (15.7.1).

!image-20231107-134319.png|width=25%!
